prisma/schema.prisma

```diff
@@ -105,3 +108,54 @@ model Attachment {
   @@index([filePath], name: "IX_Attachments_filePath")
   @@map("Attachments")
 }
+
+model ActivityLog {
+  id                   String                @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
+  createdAt            DateTime              @default(now())
+  taskId               String                @db.Uuid
+  workspaceId          String                @db.VarChar(32)
+  activityType         ActivityType
+  createTaskTracker    CreateTaskTracker?
+  assigneeTracker      AssignTracker?
+  workflowStateTracker WorkflowStateTracker?
```

Comment:
are we creating new tables for each activity log type? A JSONB column can accomodate the payload required for all the activity logs.

```
id: uuid
taskId: uuid
workspaceId: string
type: ActitvityLogType { task-created, task-updated }
data: JSONB (flexible schema)
userId: uuid (Initiator)
createdAt: timestamp
updatedAt: timestamp
```

then make a resuable service that takes required data and stores that to activity log.

We can add typesafety based on the type of activity log